DrawerKit/DrawerKit/Internal API/DrawerState.swift

```diff
@@ -0,0 +1,8 @@
+import Foundation
+
+enum DrawerState: Equatable {
```

Comment:
So compiler now can synthesize equatable?

Reply:
For enums without associated values I think it had for a while.

Reply:
No. The implementation is `internal` because it depends on a custom `equal` method that is also `internal` (actually, `private` but it will soon have to be `internal`). See the `PresentationController` file.

```
extension DrawerState {
    static func ==(lhs: DrawerState, rhs: DrawerState) -> Bool {
        switch (lhs, rhs) {
        case (.collapsed, .collapsed),
             (.partiallyExpanded, .partiallyExpanded),
             (.fullyExpanded, .fullyExpanded):
            return true
        case let (.transitioning(lhsCurPosY), .transitioning(rhsCurPosY)):
            return equal(lhsCurPosY, rhsCurPosY)
        default:
            return false
        }
    }
}
```

Reply:
The issue is that equality of floating point numbers is a tricky thing, so I have

```
private func equal(_ lhs: CGFloat, _ rhs: CGFloat) -> Bool {
    let epsilon: CGFloat = 1e-6
    return abs(lhs - rhs) <= epsilon
}
```